pkg/endpoint/policy.go

```diff
@@ -589,15 +591,6 @@ func (e *Endpoint) regeneratePolicy(owner Owner) (isPolicyComp bool, err error)
 	// Note - endpoint policy enforcement must be determined BEFORE this function!
 	e.computeDesiredPolicyMapState(repo)
 
-	// If we are in this function, then policy has been calculated.
-	if !e.policyCalculated {
```

Comment:
I think we had some code that did check the revision number but it might have been changed. @rlenglet was fixing some bug around init generation when I saw this, so he might know more. This might be used to force an initial policy update via the forceRegeneration flag.

Reply:
Yeah, @rlenglet added a wait for the endpoint to be at policy revision 1 in the code - we bumped the initial policy repository revision to 1, while the endpoint's policy revision is initialized at zero. This means that a non-zero value for the endpoint's policy revision indicates that it has been entirely regenerated (i.e. policy, BPF program, PolicyMap, proxy) successfully and has had its revision bumped. Also, I noticed that we don't even check the result of `regeneratePolicy` anymore because we always try and see if the endpoint needs to be regenerated after, so I think I can go ahead and get rid of that return value from it as well since it's somewhat misleading now.

Reply:
To clarify - the reason we don't need this anymore is because it doesn't affect what is done after we call `regeneratePolicy` - the returned value isn't used anywhere, so we aren't changing any functional behavior anymore.